hybrid/pkg/agent/agent_test.go

```diff
@@ -0,0 +1,430 @@
+			"version" : "2017-09-30"
+		}",
+		"signature":"eO4+90PuN8bZaIJjpBe1/mAzPhvSrrhLATwPFaOPzK5ZSUpsbVOuK2tXjMYkx+ora7mcaL0G45li\nbZLGUIee+DF/YZ8/5RuNf1Z8yn+5e2AqLvNhIsF5IOVZWk8yDvl/jBJCcW8GaRblldWdMoDiC2OA\nqVyRjyJCXUySNu0JADE="
+	}`
+
+// ------------------------------------------------------------------------------------------------------------------------
+
+func TestNew(t *testing.T) {
+	hybridPlugin := New()
+	require.NotNil(t, hybridPlugin, "New should return a non-nil value")
+	require.IsType(t, &HybridPluginAgent{}, hybridPlugin, "New should return a HybridPluginAgent")
+}
+
+func TestMethodsThatParseHclConfig(t *testing.T) {
+	interceptor := new(InterceptorWrapper)
+	plugin := HybridPluginAgent{interceptor: interceptor}
+
+	pluginAstNode, err := plugin.decodeStringAndTransformToAstNode(pluginsString)
+
+	require.NoError(t, err, "Error decoding test string")
+	require.Len(t, pluginAstNode, 2, "Could not transform HCL string configuration: %w", err)
+	require.Contains(t, pluginAstNode, "k8s_psat", "Could not access k8s_psat plugin by index on ast node")
+	require.Contains(t, pluginAstNode, "aws_iid", "Could not access aws_iid plugin by index on ast node")
+
+	pluginNames, pluginsData := plugin.parseReceivedData(pluginAstNode)
+
+	require.Len(t, pluginNames, 2, "Could not parse plugin names")
+	require.Contains(t, pluginNames, "k8s_psat", "Could not access k8s_psat plugin by index after parsing")
+	require.Contains(t, pluginNames, "aws_iid", "Could not access aws_iid plugin by index after parsing")
+
+	require.Len(t, pluginsData, 2, "Could not parse plugin data")
+	require.Contains(t, pluginsData, "k8s_psat", "Could not access k8s_psat plugin by index after parsing")
+	require.Equal(t, "\n  cluster = \"hybrid-node-attestor\"\n", pluginsData["k8s_psat"], "k8s_psat plugin data was not extracted properly")
+	require.Contains(t, pluginsData, "aws_iid", "Could not access aws_iid plugin by index after parsing")
+	require.Equal(t, "\n  accountId = 123456789\n", pluginsData["aws_iid"], "aws_iid plugin data was not extracted properly")
+}
+
+func TestSupportedPluginsInitialization(t *testing.T) {
+	interceptor := new(InterceptorWrapper)
+	plugin := HybridPluginAgent{interceptor: interceptor}
+
+	plugins, err := plugin.initPlugins([]string{"aws_iid", "k8s_psat", "azure_msi", "gcp_iit"})
+
+	require.NoError(t, err, "Error initializing supported plugins: %w", err)
+	require.IsType(t, &awsiid.IIDAttestorPlugin{}, plugins[0].Plugin, "Could not initialize aws_iid plugin")
+	require.IsType(t, &k8spsat.AttestorPlugin{}, plugins[1].Plugin, "Could not initialize k8s_psat plugin")
+	require.IsType(t, &azuremsi.MSIAttestorPlugin{}, plugins[2].Plugin, "Could not initialize azure_msi plugin")
+	require.IsType(t, &gcpiit.IITAttestorPlugin{}, plugins[3].Plugin, "Could not initialize gcp_iit plugin")
+
+	plugins, err = plugin.initPlugins([]string{"aws_iid_test", "k8s_psat_test"})
+	require.Error(t, err, "Error initializing supported plugins: %w", err)
```

Comment:
I've added this assetion:
require.Equal(t, err, status.Error(codes.FailedPrecondition, "please provide one of the supported plugins."))

but I think even without that it was sufficiend, because of the assert thet checks the length and spits the message that the plugin list should be 0 on unknown plugins.

Reply:
on `Equal` second parameter is expected.
However since you are using spire as dependency you can rely on spiretest package that has some utilities to compare status errors

Reply:
changed to spiretest.RequireGRPCStatus